diff --git a/src/include/duckdb/main/database.hpp b/src/include/duckdb/main/database.hpp
--- a/src/include/duckdb/main/database.hpp
+++ b/src/include/duckdb/main/database.hpp
@@ -16,6 +16,12 @@
 
 namespace duckdb {
 class LocalDatabaseFileSystem;
+
+extern bool preloaded_httpfs;
+extern bool preloaded_avro;
+extern bool preloaded_iceberg;
+extern bool preloaded_ducklake;
+extern bool preloaded_quack;
 
 class BufferManager;
 class DatabaseManager;
diff --git a/src/main/database.cpp b/src/main/database.cpp
--- a/src/main/database.cpp
+++ b/src/main/database.cpp
@@ -552,6 +552,22 @@ idx_t DuckDB::NumberOfThreads() {
 }
 
 bool DatabaseInstance::ExtensionIsLoaded(const string &name) {
+	auto extension_name = ExtensionHelper::GetExtensionName(name);
+	if (extension_name == "httpfs" && preloaded_httpfs) {
+		return true;
+	}
+	if (extension_name == "avro" && preloaded_avro) {
+		return true;
+	}
+	if (extension_name == "iceberg" && preloaded_iceberg) {
+		return true;
+	}
+	if (extension_name == "ducklake" && preloaded_ducklake) {
+		return true;
+	}
+	if (extension_name == "quack" && preloaded_quack) {
+		return true;
+	}
 	return extension_manager->ExtensionIsLoaded(name);
 }
 
