diff --git a/src/include/duckdb/main/database.hpp b/src/include/duckdb/main/database.hpp
--- a/src/include/duckdb/main/database.hpp
+++ b/src/include/duckdb/main/database.hpp
@@ -17,7 +17,9 @@
 #include "duckdb/main/extension_manager.hpp"
 
 namespace duckdb {
 
+extern bool preloaded_httpfs;
+
 class BufferManager;
 class DatabaseManager;
 class StorageManager;
diff --git a/src/main/database.cpp b/src/main/database.cpp
--- a/src/main/database.cpp
+++ b/src/main/database.cpp
@@ -510,6 +510,10 @@
 }
 
 bool DatabaseInstance::ExtensionIsLoaded(const string &name) {
+	auto extension_name = ExtensionHelper::GetExtensionName(name);
+	if (extension_name == "httpfs" && preloaded_httpfs) {
+		return true;
+	}
 	return extension_manager->ExtensionIsLoaded(name);
 }
 
